src/main/java/duke/tasks/Deadline.java

```diff
@@ -0,0 +1,73 @@
+package duke.tasks;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import duke.exception.DukeInvalidDateException;
+import duke.exception.DukeInvalidTaskException;
+
+
+/**
+ * The Deadline class is used to represent the task of a deadline nature.
+ * This object contains the task name as well as the day/date of the deadline.
+ * Inherits from Task class.
+ */
+
+public class Deadline extends Task {
+    private String date;
+    private LocalDate dateTime;
+
+    /**
+     * Initializes a Deadline object
+     *
+     * @param taskName name or description of task
+     * @param date date in which task has to be completed
+     * @throws DukeInvalidDateException
+     * @throws DukeInvalidTaskException
+     */
+
+    public Deadline(String taskName, String date) throws DukeInvalidDateException, DukeInvalidTaskException {
+        super(taskName);
+        if (!date.equals(null) && !date.equals(" ")) {
```

Comment:
Maybe using guard clauses here can ensure that your main program logic is more prominent?

![image](https://user-images.githubusercontent.com/28984700/91932375-913e5200-ed18-11ea-9a1f-292a829be8f7.png)